content/en/docs/ops/configuration/extensibility/wasm-module-distribution/index.md

```diff
@@ -8,121 +8,65 @@ aliases:
-               vm_id: basic-auth
-               runtime: envoy.wasm.runtime.v8
-               code:
-                 remote:
-                   http_uri:
-                     uri: https://github.com/istio-ecosystem/wasm-extensions/releases/download/{{< istio_version >}}.0/basic-auth.wasm
-                   # Optional: specifying sha256 checksum will let istio agent verify the checksum of downloaded artifacts.
-                   # It is **highly** recommended to provide the checksum, since missing checksum will cause the Wasm module to be downloaded repeatedly.
-                   # To compute the sha256 checksum of a Wasm module, download the module and run `sha256sum` command with it.
-                   # sha256: <WASM-MODULE-SHA>
-             # The configuration for the Wasm extension itself
-             configuration:
-               '@type': type.googleapis.com/google.protobuf.StringValue
-               value: |
-                 {
-                   "basic_auth_rules": [
-                     {
-                       "prefix": "/productpage",
-                       "request_methods":[ "GET", "POST" ],
-                       "credentials":[ "ok:test", "YWRtaW4zOmFkbWluMw==" ]
-                     }
-                   ]
-                 }
+With this configuration, an HTTP filter will be injected into ingress gateway proxies as an authentication filter.
+When the configuration reaches Istio proxy, the Istio agent will intercept the `WasmPlugin` configuration, download remote Wasm modules from the OCI image registry to a local file, and inject the HTTP filter by referencing that local Wasm module file.
+The `pluginConfig` field will be converted to the following JSON string, which will be loaded by the basic auth plugin at initialization:
+
+{{< text json >}}
+{
+  "basic_auth_rules": [
+    {
+      "prefix": "/productpage",
+      "request_methods":[ "GET", "POST" ],
+      "credentials":[ "ok:test", "YWRtaW4zOmFkbWluMw==" ]
+    }
+  ]
+}
 {{< /text >}}
 
-The Istio agent will only intercept and download remote Wasm modules configured via ECDS resources.
-This feature is enabled by default.
-To disable ECDS interception and Wasm downloading in the Istio agent, set the `ISTIO_AGENT_ENABLE_WASM_REMOTE_LOAD_CONVERSION` environment variable to `false`.
-For example, to set it globally:
-
-{{< text yaml >}}
-meshConfig:
-  defaultConfig:
-    proxyMetadata:
-      ISTIO_AGENT_ENABLE_WASM_REMOTE_LOAD_CONVERSION: "false"
-{{< /text >}}
+For more example usage of `WasmPlugin` API, please take a look at the [API reference](/docs/reference/config/proxy_extensions/wasm-plugin/).
 
 There are several known limitations with this module distribution mechanism, which will be addressed in future releases:
 
-- Envoy's extension configuration discovery service only supports HTTP filters.
-- Modules can only be downloaded through HTTP/HTTPS.
+- Only HTTP filters are supported.
+- Modules can only be fetched from public OCI image registry.
```

Comment:
FWIW, I haven't checked yet, but I guess it is actually possible if we configure `~/.docker/config.json` is placed. That is because we use the default keychain by default: https://github.com/istio/istio/blob/master/pkg/wasm/imagefetcher.go#L60